forc-plugins/forc-client/src/cmd/call.rs

```diff
@@ -0,0 +1,237 @@
+
+# Call a contract without function parameters
+» forc call 0x0dcba78d7b09a1f77353f51367afd8b8ab94b5b2bb6c9437d9ba9eea47dede97 \
+    --abi ./contract-abi.json \
+    get_name
+
+# Call a contract that makes external contract calls
+» forc call 0x0dcba78d7b09a1f77353f51367afd8b8ab94b5b2bb6c9437d9ba9eea47dede97 \
+    --abi ./contract-abi.json \
+    transfer 0xf8f8b6283d7fa5b672b530cbb84fcccb4ff8dc40f8176ef4544ddb1f1952ad07 \
+    --contracts 0xf8f8b6283d7fa5b672b530cbb84fcccb4ff8dc40f8176ef4544ddb1f1952ad07
+
+# Call a contract in simulation mode
+» forc call 0x0dcba78d7b09a1f77353f51367afd8b8ab94b5b2bb6c9437d9ba9eea47dede97 \
+    --abi ./contract-abi.json \
+    add 1 2 \
+    --mode simulate
+
+# Call a contract in dry-run mode on custom node URL using explicit signing-key
+» forc call 0x0dcba78d7b09a1f77353f51367afd8b8ab94b5b2bb6c9437d9ba9eea47dede97 \
+    --node-url "http://127.0.0.1:4000/v1/graphql" \
+    --signing-key 0x... \
+    --abi ./contract-abi.json \
+    add 1 2 \
+    --mode dry-run
+
+# Call a contract in live mode which performs state changes on testnet using forc-wallet
+» forc call 0x0dcba78d7b09a1f77353f51367afd8b8ab94b5b2bb6c9437d9ba9eea47dede97 \
+    --testnet \
+    --wallet \
+    --abi ./contract-abi.json \
+    add 1 2 \
+    --mode live
+
+# Call a contract payable function which transfers value of native asset on mainnet
+» forc call 0x0dcba78d7b09a1f77353f51367afd8b8ab94b5b2bb6c9437d9ba9eea47dede97 \
+    --abi ./contract-abi.json \
+    transfer 0xf8f8b6283d7fa5b672b530cbb84fcccb4ff8dc40f8176ef4544ddb1f1952ad07 \
+    --mode live \
+    --amount 100
+
+# Call a contract payable function which transfers value of custom asset
+» forc call 0x0dcba78d7b09a1f77353f51367afd8b8ab94b5b2bb6c9437d9ba9eea47dede97 \
+    --abi ./contract-abi.json \
+    transfer 0xf8f8b6283d7fa5b672b530cbb84fcccb4ff8dc40f8176ef4544ddb1f1952ad07 \
+    --amount 100 \
+    --asset-id 0x0087675439e10a8351b1d5e4cf9d0ea6da77675623ff6b16470b5e3c58998423 \
+    --live
+"#)]
+pub struct Command {
+    /// The contract ID to call.
+    pub contract_id: ContractId,
+
+    /// Path or URI to a JSON ABI file.
+    #[clap(long, value_parser = parse_abi_path)]
+    pub abi: Either<PathBuf, Url>,
+
+    /// The function selector to call.
+    /// The function selector is the name of the function to call (e.g. "transfer").
+    /// It must be a valid selector present in the ABI file.
+    pub function: FuncType,
+
+    /// Arguments to pass into the function to be called.
+    pub function_args: Vec<String>,
+
+    #[clap(flatten)]
+    pub node: NodeTarget,
+
+    /// Select the caller to use for the call
+    #[clap(flatten)]
+    pub caller: Caller,
+
+    /// Call parameters to use for the call
+    #[clap(flatten)]
+    pub call_parameters: CallParametersOpts,
+
+    /// The execution mode to use for the call; defaults to dry-run; possible values: dry-run, simulate, live
+    #[clap(long, default_value = "dry-run")]
```

Comment:
Is it common in other tools to default to dryrun? A user gave feedback on sway playground asking to [default to live](https://github.com/FuelLabs/sway-playground/issues/110) since it currently defaults to dry-run.

Reply:
Yeap this is common. 
Most people generally want to make queries - to query state. 
Querying state itself with live mode will cost user funds - we definitely don't want that.
Also if the user does not have funds, then they can't use live mode at all and hence can't perform any operation. 
Due to this dry-run, or simulation modes should be the default imo. 

Users who want to perform state changing operations that costs funds should be explicit (imo).